www/api/class/centreon_metric.class.php

```diff
@@ -82,9 +84,19 @@ public function getList()
         }
 
         $query = 'SELECT DISTINCT(`metric_name`)
-            COLLATE utf8_bin as "metric_name" FROM `metrics`
-            WHERE metric_name LIKE :name
-            ORDER BY `metric_name` COLLATE utf8_general_ci ';
+            COLLATE utf8_bin as "metric_name", index_id FROM `metrics` as m, index_data i
+            WHERE metric_name LIKE :name ';
+
+        /**
+         * If ACLs on, then only return metrics linked to services that the user can see.
+         */
+        if (!$centreon->user->admin) {
```

Comment:
I think it would have been better to do a private method that you would have called